### DISCUSSION TITLE
API slice onQueryStarted, getState seems to only want to access state of the API slice, intentional?

### DISCUSSION THREAD

**Discussion by julian-kingman-lark** [reduxjs/redux-toolkit#3522] [Q&A]
I have an API slice with a bit that looks like this:
```jsx
async onQueryStarted(syncData, { dispatch, getState, queryFulfilled }) {
        // @ts-expect-error it seems getState expects to only access the state of this slice
        const data = getData(getState());
```
I get the error:
```
Argument of type 'RootState<any, any, "apiSlice">' is not assignable to parameter of type 'EmptyObject & { data: DataState...
```
I _think_ this is because `getState` is expecting to only access local API slice state. Is that correct? Is that intentional? I need access to other parts of state, should I be using `store` directly? (redux tends not to like that, b/c it creates dependency circles).

Thanks in advance! I hope it becomes easier to integrate regular slices with API slices in the future.

**Comment by markerikson** [comment]
No, `getState` here will return the entire Redux store root state object (as always).

The problem is the TS typing. Because of how `createApi` works, we don't know the right TS types for the return value of `getState`.  You'll need to cast it yourself here, like `getData(getState() as RootState)`.
